### ISSUE TITLE
Airbyte Cloud: Complete email verification logic

### ISSUE THREAD

**Issue by cgardens** [airbytehq/airbyte#5449]
## Tell us about the problem you're trying to solve
* We need to set up email verification for several interactions. This got bumped after beta launch. Blocked on the BE figuring out what this should look like.

Relevant User flows: 
* [x] create user email verification (see #5214)
* [x] resend email verification (this should happen as part of #5214, since this should always trigger if someone attempts to log in and hasn't verified their email).
* [x] ~update email (verification)~
  - Won't do before cloud launch.
  - #6759
* [x] ~update email resend email verification~
  - Won't do before cloud launch.
  - #6759
* [x] ~invite user (See #5455 & #6486)~
* [x] ~ability to resend an invitation to a user~
  - Won't do before cloud launch.
  - #6760

**Comment by tuliren** [comment]
> - update email (verification)
> - update email resend email verification

Currently there is no way to change email.

**Comment by tuliren** [comment]
> - resend invitation

- Clicking the `resend` link results in the following error in console:

<img width="560" alt="Screen Shot 2021-09-28 at 13 20 35" src="https://user-images.githubusercontent.com/1933157/135159953-78ac7add-b0a0-40bf-b42b-7ceac472cbe4.png">

- On the verify email page, the `Back` button on the top left corner does not work.
  - https://cloud.airbyte.io/confirm-verify-email

**Comment by tuliren** [comment]
TODO:
- Error message
  - Show error message on UI.
- Email update
  - Low priority
  - Check if Firebase can handle email update first.
  - Update email only when user clicks the verification link. In this way, people won't be logged out after they change the email and before they complete the verification.
